app/scripts/controllers/pod.js

```diff
@@ -117,6 +180,9 @@ angular.module('openshiftConsole')
               $scope.pod = pod;
               setLogVars(pod);
               setContainerVars();
+
+              $scope.containerTerminals = makeTerminals($scope.containerTerminals);
```

Comment:
You should only need to call `makeTerminals` once in the list and just update the container statuses here. The list of containers in the pod won't change. It would simplify `makeTerminals` as well since you don't have to check if the terminal already was created.